csharp/src/Ice/OutputStream.cs

```diff
@@ -1137,19 +1137,38 @@ internal void RewriteInt(int v, Position pos)
                 $"offset: {pos.Offset} segment size: {Size - _segmentList.Take(pos.Segment).Sum(data => data.Count)}");
             Span<byte> data = stackalloc byte[4];
             MemoryMarshal.Write(data, ref v);
+            RewriteSpan(data, pos);
+        }
+
+        internal void RewriteSize(int size, Position pos)
```

Comment:
I think all the Rewrite methods can and should be private, with the exception (for now) of RewriteSize.